tests/Findologic/Response/Xml21ResponseParserTest.php

```diff
@@ -271,6 +270,65 @@ public function testFiltersAreReturnedAsExpected(Xml21Response $response, array
         $this->assertEquals($expectedFilters, $filters);
     }
 
+    public function smartSuggestBlocksProvider()
+    {
+        return [
+            'No smart suggest blocks are sent and category filter is not in response' => [
+                'demoResponse' => 'demoResponseWithoutCategoryFilter.xml',
+                'flBlocks' => [],
+                'expectedFilterName' => null,
+                'isHidden' => null
+            ],
+            'Smart suggest blocks are sent and category filter is not in response' => [
+                'demoResponse' => 'demoResponseWithoutCategoryFilter.xml',
+                'flBlocks' => ['cat' => 'Category'],
+                'expectedFilterName' => 'Category',
+                'isHidden' => true
+
+            ],
+            'No smart suggest blocks are sent and category filter is available in response' => [
+                'demoResponse' => 'demoResponseWithCategoryFilter.xml',
+                'flBlocks' => [],
+                'expectedFilterName' => 'Kategorie',
+                'isHidden' => false
+
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider smartSuggestBlocksProvider
+     */
+    public function testSmartSuggestBlocks(
```

Comment:
`testSmartSuggestBlocks` does not really tell you what this actually tests. Try to find something more descriptive.